oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/endpoint/AbstractOAuth2AuthorizationGrantRequestEntityConverter.java

```diff
@@ -69,6 +71,16 @@
 	 */
 	abstract MultiValueMap<String, String> createParameters(T authorizationGrantRequest);
 
+	/**
+	 * Sets the flag that controls whether client credentials are encoded using the
+	 * application/x-www-form-urlencoded algorithm in the headers converter.
+	 * @param encodeClientCredentials {@code true} to encode client credentials (default
+	 * is false)
+	 */
+	public void setEncodeClientCredentials(boolean encodeClientCredentials) {
```

Comment:
Remove setter as this bug fix is scheduled for `5.5.1` so we want this to be a patch update.

As well, the client identifier and client secret MUST be encoded using application/x-www-form-urlencoded, as per [spec](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1):

> The client identifier is encoded using the
   "application/x-www-form-urlencoded" encoding algorithm per
   Appendix B, and the encoded value is used as the username; the client
   password is encoded using the same algorithm and used as the
   password.